src/pushNotifications/WasteReminder.ts

```diff
@@ -0,0 +1,126 @@
+import * as SecureStore from 'expo-secure-store';
+
+import { device, secrets, staticRestSuffix } from '../config';
+import * as appJson from '../../app.json';
+import { getPushTokenFromStorage, PushNotificationStorageKeys } from './TokenHandling';
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+
+type SettingInfo = {
+  city: string;
+  reminderTime: string;
+  onDayBefore: boolean;
+  street: string;
+  wasteType: string;
+  zip: string;
+};
+
+export const updateReminderSettings = async ({
+  city,
+  reminderTime,
+  onDayBefore,
+  street,
+  wasteType,
+  zip
+}: SettingInfo) => {
+  const accessToken = await SecureStore.getItemAsync('ACCESS_TOKEN');
+  const pushToken = await getPushTokenFromStorage();
+  const requestPath = secrets[namespace].serverUrl + staticRestSuffix.wasteReminderRegister;
+  const os =
+    device.platform === 'ios' || device.platform === 'android' ? device.platform : 'undefined';
+
+  const fetchObj = {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      Authorization: 'Bearer ' + accessToken,
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      waste_registration: {
+        notify_at: reminderTime,
+        notify_for_waste_type: wasteType,
+        street,
+        city,
+        zip,
+        notify_days_before: onDayBefore ? '1' : '0'
+      },
+      notification_device: {
+        token: pushToken,
+        device_type: os
+      }
+    }),
+    cache: 'no-cache'
+  };
+
+  if (accessToken && pushToken) {
+    try {
+      const response = await fetch(requestPath, fetchObj);
+      const json = await response.json();
+
+      return json?.id as number | undefined;
+    } catch {
+      return undefined;
+    }
+  }
+};
+
+export const getReminderSettings = async () => {
+  const accessToken = await SecureStore.getItemAsync('ACCESS_TOKEN');
+  const pushToken = await SecureStore.getItemAsync(PushNotificationStorageKeys.PUSH_TOKEN);
+  const requestPath =
+    secrets[namespace].serverUrl + staticRestSuffix.wasteReminderRegister + `?token=${pushToken}`;
+
+  const fetchObj = {
+    headers: {
+      Authorization: 'Bearer ' + accessToken,
+      'Content-Type': 'application/json'
+    },
+    cache: 'no-cache'
+  };
+
+  if (accessToken && pushToken) {
+    return fetch(requestPath, fetchObj)
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+      })
+      .catch(() => {
+        return undefined;
+      });
```

Comment:
why `.then` and `.catch` here and in the method before the `await`-way? is there a specific reason for handling it differently?

Reply:
i initially had no `await` syntax at all here, and changed it to that for the update function specifically, because there was a need of also awaiting and the `json()`promise and do an additional step with the result.

at that point it was more readable to switch to the `await` syntax.